tests/unit/test_buffer_handling.py

```diff
@@ -95,6 +94,82 @@ def test__get_sparse_buffer_properties(component_type, is_columnar):
         assert properties.columns is None
 
 
+@pytest.mark.parametrize(
+    ("component_type", "is_columnar"),
+    [
+        pytest.param(ComponentType.sym_load, True, id="sym_load-single-columnar"),
+        pytest.param(ComponentType.sym_load, False, id="sym_load-single-row_based"),
+        pytest.param(ComponentType.asym_load, True, id="asym_load-single-columnar"),
+        pytest.param(ComponentType.asym_load, False, id="asym_load-single-row_based"),
+    ],
+)
+def test__get_buffer_properties__batch_requested_for_single_data(component_type, is_columnar):
+    data = load_data(component_type, is_batch=False, is_columnar=is_columnar, is_sparse=False)
+    schema = power_grid_meta_data[DatasetType.update][component_type]
+
+    with pytest.raises(
+        ValueError, match="Incorrect/inconsistent data provided: single data provided but batch data expected."
+    ):
+        get_buffer_properties(data, schema=schema, is_batch=True, batch_size=BATCH_DATASET_NDIM)
```

Comment:
I see it doesn't originate from this PR, but can you make the load_data also use the constants like BATCH_DATASET_NDIM?

Everything else LGTM.